pkg/olm/catalogsource_test.go

```diff
@@ -0,0 +1,327 @@
+			if testCase.expectedError == "" {
+				assert.NotNil(t, catalogSource.Definition)
+				assert.Equal(t, testCase.name, catalogSource.Definition.Name)
+				assert.Equal(t, testCase.namespace, catalogSource.Definition.Namespace)
+			}
+		} else {
+			assert.Nil(t, catalogSource)
+		}
+	}
+}
+func TestPullCatalogSource(t *testing.T) {
+	generateCatalogSource := func(name, namespace string) *oplmV1alpha1.CatalogSource {
+		return &oplmV1alpha1.CatalogSource{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: namespace,
+			},
+			Spec: oplmV1alpha1.CatalogSourceSpec{
+				Image: "test",
+			},
+		}
+	}
+
+	testCases := []struct {
+		name                string
+		namespace           string
+		addToRuntimeObjects bool
+		expectedError       error
+		client              bool
+	}{
+		{
+			name:                "catalogsource",
+			namespace:           "test-namespace",
+			addToRuntimeObjects: true,
+			expectedError:       nil,
+			client:              true,
+		},
+		{
+			name:                "",
+			namespace:           "test-namespace",
+			addToRuntimeObjects: true,
+			expectedError:       fmt.Errorf("catalogsource 'name' cannot be empty"),
+			client:              true,
+		},
+		{
+			name:                "catalogsource",
+			namespace:           "test-namespace",
+			addToRuntimeObjects: false,
+			expectedError:       fmt.Errorf("catalogsource object catalogsource does not exist in namespace test-namespace"),
+			client:              true,
+		},
+		{
+			name:                "catalogsource",
+			namespace:           "test-namespace",
+			addToRuntimeObjects: true,
+			expectedError:       fmt.Errorf("catalogsource 'apiClient' cannot be empty"),
+			client:              false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		// Pre-populate the runtime objects
+		var runtimeObjects []runtime.Object
+
+		var testSettings *clients.Settings
+
+		catalogSource := generateCatalogSource(testCase.name, testCase.namespace)
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, catalogSource)
+		}
+
+		if testCase.client {
+			testSettings = clients.GetTestClients(clients.TestClientParams{
+				K8sMockObjects: runtimeObjects, SchemeAttachers: testSchemes})
+		}
+
+		builderResult, err := PullCatalogSource(testSettings, testCase.name, testCase.namespace)
+		assert.Equal(t, testCase.expectedError, err)
+
+		if testCase.expectedError == nil {
+			assert.Equal(t, testCase.name, builderResult.Object.Name)
+			assert.Equal(t, testCase.namespace, builderResult.Object.Namespace)
+		}
+	}
+}
+
+func TestCatalogSourceGet(t *testing.T) {
+	testCases := []struct {
+		catalogSource *CatalogSourceBuilder
+		expectedError error
+	}{
+		{
+			catalogSource: buildValidCatalogSourceBuilder(buildTestClientWithDummyObject()),
+			expectedError: nil,
+		},
+		{
+			catalogSource: buildInValidCatalogSourceBuilder(buildTestClientWithDummyObject()),
+			expectedError: fmt.Errorf("catalogsource 'nsname' cannot be empty"),
```

Comment:
IMO it's more valuable to test Get with the resource does and doesn't exist. Otherwise, you're just testing NewBuilder